### DISCUSSION TITLE
How to quickly remove duplicate lines?

### DISCUSSION THREAD

**Discussion by borifax** [WinMerge/winmerge#1131] [Q&A]
Does anyone know a way to quickly remove duplicate lines from one of the two files?
After comparing the two files, I got the result as shown in the screenshot and I don't know how to efficiently remove repeated lines from one of them.

![image](https://user-images.githubusercontent.com/45988475/103081828-0d66e900-45d9-11eb-9794-4912874188bb.png)

**Comment by sdottaka** [comment]
WinMerge doesn't have the ability to remove duplicate rows, but you can do something similar with regular expression substitutions.

![image](https://user-images.githubusercontent.com/98126/103143289-70ca5580-4756-11eb-90ab-cf2b2fde6388.png)

Or, a general text editor has such a function, so please try using it.

For example, in Notepad ++, see the following URL.

https://stackoverflow.com/questions/3958350/removing-duplicate-rows-in-notepad

**Comment by borifax** [comment]
Ok, but this method only works for duplicate lines in one file.
The ability to sort by color would solve the problem.

**Comment by sdottaka** [comment]
To sort with WinMerge, select the lines you want to sort and select Plugin-> Script-> SortAscending or SortDescending menu item.

A typical editor also has such a feature.

**Comment by borifax** [comment]
We don't understand each other. How can I select only the white lines from the right window?

![image](https://user-images.githubusercontent.com/45988475/103345639-39c7ad00-4a92-11eb-8cd2-6e4981044508.png)

**Comment by sdottaka** [comment]
I don't think I understand exactly what you want
 but with the commit https://github.com/WinMerge/winmerge/commit/a55321928259a8b541ecb1fafe423787f781bbe2, View-> Diff Context-> Invert A menu item has been added that allows you to see only the identical lines and select and delete only the identical lines.

![issue520](https://user-images.githubusercontent.com/98126/103407660-cb264480-4ba2-11eb-9c01-6c7acf1d41d4.gif)

**Comment by borifax** [comment]
I have two documents.
1) I compare them. WM finds the differences. But:
2) I need to delete from one of these documents all those rows which are defined as doubled with rows from the other document.

As I can see that the new feature you just added solves the problem.
Thank you.
